Dshop/apps/users/tests/test_api_login.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+
+User = get_user_model()
+
+
+@pytest.fixture
+def api_client():
+    return APIClient()
+
+
+@pytest.fixture
+def login_url():
+    return reverse('api-login')
+
+
+@pytest.fixture
+def login_data():
+    return {
+        'username': 'testuser',
+        'password': 'testpassword',
+    }
+
+
+@pytest.fixture
+def user_instance(login_data):
+    return User.objects.create_user(**login_data)
+
+
+@pytest.mark.django_db
+def test_login_success(api_client, login_url, login_data, user_instance):
+    response = api_client.post(login_url, login_data, format='json')
+    assert response.status_code == status.HTTP_200_OK
+
+
+@pytest.mark.django_db
+def test_login_empty_data(api_client, login_url, user_instance):
+    response = api_client.post(login_url, {}, format='json')
+    assert response.status_code == status.HTTP_400_BAD_REQUEST

```

Comment:
Please add assert for error fields. 
From this test I see it returns 400 error, and I can assume that it is because of empty data. 
Adding assert to check error fields (not messages) will clearly show that it is because of the missing data. 
Please add this to all tests. 
Example
```
assert len(response.errors) == 2
assert "login" in response.errors
```
I am writing from memory so most likely this is not correct code.
Documentation: https://www.django-rest-framework.org/api-guide/exceptions/#exception-handling-in-rest-framework-views -> search for "An example validation error might look like this:"